### ISSUE TITLE
Any testers I can borrow?

### ISSUE THREAD

**Issue by lefnire** [HabitRPG/habitica#285]
@Pandoro @horusofoz @hookang @wc8 @Neohuman @SlappyBag  , and anyone else who would be so kind - I'm setting up a staging server for testing new big changes. #237 is the one I hope I fixed - would anyone mind playing around at http://habitrpg.herokuapp.com?
- Lvl 3 unlocks Parties, but it's non-functional presently - don't worry about that yet
- I added an "Emulate Next Day" button bottom right to fake cron, that'll be useful to test
- #220 (preen database) is in effect - big one to test, gotta make sure there's no data-loss as a result

**Comment by Pandoro** [comment]
Could it be the case that I cannot play with my normal account? I can simply login, but pretty much nothing is working for me. 

The emulate next day button also works only once every several tries. Should I play on a fresh account? 

EDIT: In fact I tested this as well, I'm getting the same behaviour on there as well. Checking stuff does not do anything, next day button doesn't work. 

Somehow I've got the feeling I'm totally saying this in a negative way :x It isn't supposed to sound like that! I just want you to know what I am experiencing :) Btw thanks for the invites!

**Comment by lefnire** [comment]
Well there we go! A whole host of things to debug there, Alex

**Comment by lefnire** [comment]
oh yeah, none of the interactions work now it's deployed. wonder if it's a production thing, they're doing ok on local. Ok then, temporary hold on this ticket until I get that panned out

**Comment by Pandoro** [comment]
Oooh my :( I always feel bad when I report another bug to you! Btw I was just playing around with some new armors, I was thinking of maybe sniffing for some inspiration in old game sprites. Of course we have to be a bit careful regarding copyright infringement, but it can serve as a really nice starting point I think. Starting from scratch somehow feels rather hard :-/

**Comment by lefnire** [comment]
yeah, starting from scratch has gotta be tough. Actually, look into copyright. I know things like RPG Maker use SNES iconsets, don't know how they get around copyright - do they just pay royalties? Are these assets now publicly accessible?

If we're allowed to use them, a few of those could be fun easter-eggs / special items, for the fans who recognize specifics. 1000GP for the Masamune!

**Comment by Pandoro** [comment]
I think if you want to play it official you have to ask for terms before using it. Or wait until the owner is dead for 70 years. However from what I am seeing right now, the snes icons had a higher resolution, they might serve as an inspiration, however they cannot be used. So as long as we don't market the stuff with the names of the games the pixel art was inspired from, it should be fine.

**Comment by StanLindsey** [comment]
I've added the above link to my bookmark bar so I'll be testing it regularly. =D

**Comment by dbueller** [comment]
I'd save up for the Masamune and Murasame!

**Comment by lefnire** [comment]
@Sabin144 Right?? Right??

**Comment by Pandoro** [comment]
... Yesterday  night I was playing with a pixel editor on my tablet. You have no idea how hard it is to fit a nicely curved katana into 12x12 pixels ;) Let alone making a fancy Masamune or Murasame of it! But let's see what other artists can come up with :)

**Comment by lefnire** [comment]
ok, round two - ignore the fact that there are no notifications

**Comment by Pandoro** [comment]
Okay, so the negative gold bug is still there, which comes in very handy. Overall everything seems fine, the next day button is a bit laggy, but I guess that is not really a big issue here. 

One thing I found though: 
After getting to level 3 and buying Everything, I end up with two empty placeholders. Then I died, so far so good, I got back to level one, emulated the next day and then I had two empty placeholders again + I was wearing the final gear. So I think somehow the character is not correctly reset to zero once it dies. (Apart from the fact wether this is wanted or not.) 

EDIT: I forgot the mention, it is not a visual thing only, since I loose 3 hp on the day change, while I have 4 dailies, so the less damage taken from the armor is in effect. 

Furthermore what I forgot to mention, once I put in asfs in a gold field for the rewards, I went back to zero gold and I didn't earn anything until I went to the next day. I presume it was set to NaN which is fixed in a sanity check of the cron job? But I also think that this was known.

**Comment by lefnire** [comment]
Awesome, good find! Thanks

**Comment by Pandoro** [comment]
Well on my second death this doesn't seem to occur. However after buying everything again, I lost it after one day change and it returned after another.

So it might as well be something with the day changes. 
- I got to this screen :

Application Error

An error occurred in the application and your page could not be served. Please try again in a few moments.

If you are the application owner, check your logs for details.

**Comment by lefnire** [comment]
Yep. 1000ms bundle timeout. Exactly what I was trying to avoid, back to the
lab

**Comment by Pandoro** [comment]
Aaahw man >.< :(   

I wish I could help you here! Sadly my "skills" if you can call em that are limited to java and c++ :(

So what I figured out now, the next day button sometimes simply doesn't work. I think when it has trouble reaching the server, at least this little offline reconnect button flashes every time I click it. Apart from that now everything seems to work very fine. 

Oooh and I couldn't reroll. It says I have 8 tokens, but if I click reroll, nothing happens.

**Comment by lefnire** [comment]
Reroll and buy tokens are ok, that's private branch I haven't merged.
Things not sticking or taking too long is bad - I was watching tail log
while you played around, kept triggering bundle timeout, worse that before
(which is specifically what these changes were designed to fix) . I'm
really glad we staged this, thanks for testing Alex, will dive into soon.

**Comment by Pandoro** [comment]
No problem :) Glad to help! 

Iḿ actually testing something myself. I was thinking about maybe animations for these "epic" items the kickstarter users might gonna get. Somehow gimp is messing up the transparency in gif files, but this might be something cool to have :) You know, glowing swords ^^
![test2](https://f.cloud.github.com/assets/1872333/119924/3cce2004-6cbc-11e2-8e62-a01b832517bb.gif)

EDIT: I figured out what was going wrong. Apparently gif cannot handle semi-transparent things. So while I can do animations that are transparent, I cannot do some nice overlapping glow. This sucks a little bit, considering you would need to know the background color for a nice glow outside of opaque pixels. However a shimmering diamond would be very well possible. (If such things are desirable that is ;) )

**Comment by horusofoz** [comment]
I logged in, creating a bunch of test habits, dailys and todos. Completed all todos, created progress on some habit then ticked of all the todos. Avatar progressed a level. I created a bunch more duplicate test todos then logged out and back in. The original test todos are all back.

Setup
- Test account I created so starting from scratch
- Firefox/Win7

**Comment by horusofoz** [comment]
Though notifications are not working elsewhere, they do for digital item purchases.

**Comment by lefnire** [comment]
Awesome, you guys are great

**Comment by horusofoz** [comment]
Ten attempts at "Emulate Next Day" failed in a row :( Firefox/Win7

**Comment by horusofoz** [comment]
Can someone give me their UUID for testing the group feature?

Mine is 08099532-cdb9-4dc6-8cd4-2e104ea0dd26

**Comment by lefnire** [comment]
Just doesn't work? Or reloads slow or app crash? Ps, don't worry these
aren't terrible issues - I know mostly what's going on, you guys are
providing great info

**Comment by horusofoz** [comment]
- Second log out and log back in was not able to reproduce the issue with Todos returning.
- Notifications for anything other than digital item purchases do not appear. That includes losing HP, gaining XP/GP and purchased custom rewards. 
  Notifications for when things such as store is unlocked do display.

**Comment by wc8** [comment]
Die by purchasing rewards and you level up.

**Comment by wc8** [comment]
Looking a little further, you are told you level up, although otherwise I don't see a sign of it.

**Comment by wc8** [comment]
Apparently the level up announcement appears every time after death. Sorry if this is expected behavior. 
Is it the API token or the UUID we're to share? (I created a test account.)

**Comment by lefnire** [comment]
Uuid. The api token is meant now to safeguard your account so you can share
your uuid

**Comment by wc8** [comment]
@horusofoz ae05a82c-829f-46bf-9bfa-ce0e8d3b06cd

**Comment by lefnire** [comment]
Doesn't yet add bidirectional

**Comment by horusofoz** [comment]
Funnily enough the test site has been up without issue but I can't access prod for a couple hours :D

**Comment by wc8** [comment]
I can't access main site either.

**Comment by horusofoz** [comment]
Hmm I added @wc8 and the page refreshed but no changes. @wc8 If you add my UUID do you get any visual change to show you're part of a party?

08099532-cdb9-4dc6-8cd4-2e104ea0dd26

**Comment by wc8** [comment]
I added, but I think 1st post mentions parties not-functional yet, just unlocked.

**Comment by wc8** [comment]
I suppose the best way to test #220 would be to create a lot of "days" and data?

**Comment by lefnire** [comment]
Yep, not ready yet

**Comment by wc8** [comment]
(Off-Topic, but after a few instances of dying and leveling up, it might be a nice little way of partially addressing the concerns in #223). Meanwhile I'm basically trying to throw every curveball (same named tasks, tasks with strange symbols #, %, etc.) at the tester that I can think of. So far, there have been a couple of instances in which I thought I completed every task, but experienced HP loss, but I can't be proof-positive that I didn't miss a box and I can't replicate it when I try to. Otherwise, I haven't noticed issues.

**Comment by horusofoz** [comment]
Eek overlooked that.

**Comment by wc8** [comment]
Ok, the HP loss with completed todos seems more likely after items are purchased, swords, etc.

**Comment by wc8** [comment]
Correction: the HP loss with completed **daily tasks** seems more likely after items are purchased, swords, etc.

**Comment by horusofoz** [comment]
Main site back up :)

**Comment by horusofoz** [comment]
RE: Emulate Next Day Not Working
- Tested again in Firefox. 
- Emulate Next day still not resetting Dailys or impacting Todos.
- Tested in Chrome.
- Worked as expected first time.
- Tested again in Firefox.
- Now working as expected (Dilys reset, Todos impacted).

Can't think of anything that would cause this behaviour other than an isolated/temp DB write issue when I was originally using FF but reporting for reference.

**Comment by StanLindsey** [comment]
I've been testing it for the last hour and to my eyes everything is working properly. Dailys reset, health depletes, level ups work etc.

**Comment by horusofoz** [comment]
Can't access it any more at my end. Received the not so informative error below ;)
![Image 1](https://f.cloud.github.com/assets/3202339/121511/9cd801dc-6dab-11e2-9881-a09e636ba057.png)

**Comment by wc8** [comment]
I got the same a while ago, but it's up now.

**Comment by horusofoz** [comment]
Confirmed. back up. That was quick. Thanks for the heads up @wc8.

**Comment by horusofoz** [comment]
Hmm notification for health lost due to uncompleted dailys displayed but still no notifications show for XXP gained or HP lost on negative habit clicks.

**Comment by lefnire** [comment]
Consider it dead now my friends, y'all pointed out the errors I was looking
for (but couldn't reproduce without your help) - I'll chime in again later
on this ticket with the next attempt. Thanks so much all!

**Comment by horusofoz** [comment]
I think we can close this thread for now. @lefnire can open and comment when looking for the next round of testing.

**Comment by lefnire** [comment]
alright my friends, round 3!

(you're right jeff, i should close between)

**Comment by lefnire** [comment]
ugh. one sec.

**Comment by dbueller** [comment]
I thought there was supposed to be a confirm on delete.  I don't see it there.

Edit: Maybe I just imagined that feature was live.  Sorry.

**Comment by Pandoro** [comment]
First of all, the default dailies have all the days greyed out, yet still
they affect the general play. It get's really messy when you start turning
days on and off again.

Secondly, could you elaborate how the next day button works? Somehow if I
switch of Sunday, it is greyed out, and it stays this way, but I got the
strong feeling, the next day button does actually emulate different week
days as well. Meaning, if you have one task that is not active, on sunday
it keeps greyed out, but on switching you get damage in many cases.

Thirdly deleting todos and rewards seems not to ask for a confirmation,
this is the case however for habits and dailies.

**Comment by Pandoro** [comment]
Okay... for some reason I need to grey out Saturday, not sunday, then the
daily isn't actually grey itself, but I don't get any damage from the task.
However sometimes I simply don't get damage on uncompleted dailies :(

**Comment by lefnire** [comment]
Wait wait, no more testing - i missed something, i'll reopen (sorry)

**Comment by Pandoro** [comment]
Shoot sorry, I hope I didn't annoy the server with all my clicking :x

**Comment by StanLindsey** [comment]
Angry server is angry

**Comment by Pandoro** [comment]
![angry_server](https://f.cloud.github.com/assets/1872333/122312/0eee3ab4-6e49-11e2-8db9-3c7e0f73cef7.png)

Just to clarify, this is nothing serious, just something I had to whip up in 15 minutes after I saw your comment @SlappyBag

**Comment by horusofoz** [comment]
@Pandoro We could use this as part of the server down message.maybe have it going up and down (hammering) on the HRPG avatar?

**Comment by Pandoro** [comment]
I was thinking of something like that as well, but a sad server might be more suited than an angry server for that?

**Comment by StanLindsey** [comment]
Players used "mass login" on Server
Critical Hit!
Server is Ko'd

**Comment by lefnire** [comment]
It's perfect :)

**Comment by lefnire** [comment]
round 4! (ignoring again the notifications)

**Comment by Pandoro** [comment]
I'm at the university right now, so I don't want to annoy people with thousands of clicks, but so far it seems to work pretty much okay.

The offline reconnect thing is still happening on emulate next day, but I'm not sure if this should be the case or not. Occasionally, there is some kind of server lag, or maybe it is something else, but for some reason I couldn't buy poitions or check stuff. Then I waited a little and it worked again. 

The default tasks are still turned off for each day by default, at least this is how it looks, but they do subtract hp when not checking them.

**Comment by horusofoz** [comment]
@lefnire Testing specific a feature or general functionality?

**Comment by horusofoz** [comment]
App Error when trying to access the DEV/UAT env.

**Comment by lefnire** [comment]
@Pandoro awesome, thanks man! offline-reconnect is ok as a one flicker on cron (emulate next day) - it's the code refreshing reseting the DOM bindings. Default days on tasks is another issue, just created a ticket: #307 - thanks! Lag would be an issue, I'll play around and see what that could be

@horusofoz general functionality. crap, Heroku didn't log the App Error. Was that just from trying to load the page?

**Comment by Pandoro** [comment]
Sometimes the hp potions don't work. In one case it even took 20hp away from me just after level up :o

**Comment by horusofoz** [comment]
@lefnire The error came after 30 seconds or so of trying to load http://habitrpg.herokuapp.com

**Comment by lefnire** [comment]
Gotcha. Hmm.. Well, we'll keep it closed then till the next round - we're
very close though

**Comment by horusofoz** [comment]
It's accessible for me now. Was getting the error about 6-7 hours ago.

**Comment by lefnire** [comment]
Alright, we're looking _much_ better now that the batch-update branch is pushed live. I really wanted to test it more, but we got a surge from Facebook (sounds like some people started a wave!) which was causing the 1000ms error every few seconds, and the app was perpetually restarting. I had to go into emergency mode and push the batch-updates branch.

We went from average `8mb netOut/sec` to `110kb /sec`. That's right, we were averaging 8mb / second with ~50 concurrent users, because we can't run `db.users.update` via Racer! (see #80 #165 #237 https://github.com/codeparty/racer/issues/69) @switz check out my super-super [hack](https://github.com/lefnire/habitrpg/blob/master/src/app/schema.coffee#L65) for batching a Mongo update query (including my own forked [racer](https://github.com/lefnire/racer) _and_ [racer-db-mongo](https://github.com/lefnire/racer) - it's really quite a nightmare).

So everyone keep an eye out everyone, I ran a backup and I have the old codebase ready to re-deploy if we have to.

**Comment by switz** [comment]
yowza! nice job!

**Comment by lefnire** [comment]
ok, hopefully one more round my friends. http://habitrpg.herokuapp.com - this includes the party system (lvl 3). It's quite choppy, and just about every party action requires a refresh presently, but as long as it's not "buggy" I say we roll out with the v1. Secondly, it includes a hefty schema migration with many bug fixes, including some auth fixes and REST v1, so as long as this doesn't crash on anyone tomorrow's roll-out will be a good one.

@Pandoro since -GP is fixed, I added a debug button bottom right.

**Comment by lefnire** [comment]
_refresh_, not restart

**Comment by horusofoz** [comment]
- Had to recreate tester account as preened.
- First time I "Emulated Next Day", I lost all tasks. Refresh resolved.
- Spammed habits to try and get to level 3 quick and after X number of clicks it stopped responding as did all other habits. Emulated Next Day and still unrepsonsive. 
- Dailys have not reset. They've returned to their normal shading but are ticked and when unticked or reticked there is no change to GP/HP/XP.
- Refreshed and now functioning correctly.
- Spammed Habits for L3/Parties.
- Created HabitRPG Testers party.
- Tried adding @wc8 using previous UUID but not valid.
- Clicked off party add screen but now cannot access feature from user menu.
- Refresh restored party option.
- Can someone add me to their party for further testing? 
- UUID: 54fe8cf2-a0ba-4059-844b-279d851829d7

**Comment by switz** [comment]
Tyler, what's your userid? Trying to test out the party system.

**Comment by Pandoro** [comment]
:D me too, me too :D 

08056cfd-d6cb-4d4d-91e8-b6a27dc02a79

**Comment by horusofoz** [comment]
@Pandoro I've sent an invite to you for a party.

@switz Can you add myself (54fe8cf2-a0ba-4059-844b-279d851829d7) & @Pandoro (08056cfd-d6cb-4d4d-91e8-b6a27dc02a79) lets see what happens if you ar ea member of multiple parties.

**Comment by Pandoro** [comment]
Not possible ;)

**Comment by switz** [comment]
User already in a party or pending invitation.

**Comment by horusofoz** [comment]
OK I can see @Pandoro's avatar. @switz Can you provide your UUID and then let's perform actions and see what is/isn't working.

@lefnire Any specific functions to test or just see what we can break?

**Comment by horusofoz** [comment]
Guys lets add each other on Google+ so we can get some real time chat for testing happening. I'm interested to see what happens when we're on different days and if one of us dies or rerolls.

**Comment by StanLindsey** [comment]
Daymne I didn't get in on the party action. I did try and test other aspects but all was working fine =D

**Comment by Pandoro** [comment]
The emulate exp and gold button does not level you up, and a few seconds
ago there was a gui glitch. The hp and exp bar got very long and suddenly
there were two party members. After the next day the second member
disappeared though and the hp bar became normal again.

Got to get back to get back to work now though :x

**Comment by horusofoz** [comment]
@SlappyBag Still trying to get us all connected on G+ so we can collaborate for party testing. Jump in it.

**Comment by horusofoz** [comment]
Habit progress doesn't seem to scale. e.g. Keep clicking and it stays at a static 1.2 GP reward

**Comment by Pandoro** [comment]
This I cannot confirm, do note though that the pure + or - stats do not
scale I think.

**Comment by dbueller** [comment]
I made a party then left it.  I started another party, but I just clicked away from it and didn't complete it.  Now I have an "anonymous" lvl (blank) guy with a missing weapon image.  Also, party is missing from the settings.  It resolves on refresh.

**Comment by ghost** [comment]
Chromium Ubuntu 12.10 64bit 
Hitting the Emulate Next Day button cause the app to to refresh and disconnect. Party function not working at all. User id fe4ebe85-137c-4e6a-b8a4-9d117b02310c.

**Comment by lefnire** [comment]
Sounds like maybe I'll have to suppress parties for this next push. Thanks
for the feedback guys, I'll start toying around with those issues. Most of
them are due to model subscription issues - I'm going to try to move to
fetches instead of subscribes.

@Pandoro those UI glitches are resizing your screen area due to party
addition, I'll probably need some css help from someone in that area

@horusofoz my biggest concerns are:
- You had to recreate tester account. All production data should have
  stayed intact, I ran a migration against the database and if any data was
  lost I need to revisit the migration. Are you talking about a test account
  created specifically on the old test run, or your did your main account get
  lost? Anyone else experience any data loss from their live production
  accounts? You mentioned trying to add @wc8's previous uuid showing invalid,
  @wc8 did you lose your main account?
- next-day you lost all tasks - I'll be debugging this first. I think the
  "next day" button generally has issues, and once i get this pushed I'll
  give the google-group people some "cheat codes" - I had to wait till this
  is pushed as it does some accessControl lock down to prevent cheatcodes
  gone awry. But yeah, losing all tasks shouldn't be one of those issues.
  Thanks for spotting that.

**Comment by wc8** [comment]
@lefnire may have simply forgotten my test account password, but I couldn't log in. I created a new test account testwc8 (and recorded the password)
@horusofoz new User ID (for testwc8): 086e1c9c-8d95-4aa4-b5dd-3c968cc85c14
I missed the update in my emails so I haven't tested much yet.

**Comment by lefnire** [comment]
But your main account works ok? (site is a clone of prod)

**Comment by wc8** [comment]
My main login works fine.

**Comment by StanLindsey** [comment]
I was logging in on my normal account through Facebook and that all worked fine. I realise there are two databases working independently of one another.
